botocore/credentials.py

```diff
@@ -1321,6 +1328,18 @@ def _validate_source_profile(self, parent_profile, source_profile):
                 )
             )
 
+        profile = profiles[source_profile]
```

Comment:
I was originally a little confused because `source_profile`/`parent_profile` refer to the name of the profile while `profile` refers to the configuration and I made the assumption they would all be the same type. I wonder if it makes sense to just not create the `profile` variable and use `profiles[source_profile]` directly or call the variable something like `source_profile_config` or rename the other two parameters to `source_profile_name` and `parent_profile_name`?

Reply:
Yeah, I'll see what I can do about this.